plugin/pkg/shoot/managedseed/admission.go

```diff
@@ -180,13 +181,35 @@ func (v *ManagedSeed) validateUpdate(ctx context.Context, a admission.Attributes
 		allErrs = append(allErrs, field.Invalid(field.NewPath("spec", "kubernetes", "enableStaticTokenKubeconfig"), shoot.Spec.Kubernetes.EnableStaticTokenKubeconfig, "shoot static token kubeconfig cannot be disabled when the seed secretRef is set"))
 	}
 
+	zoneValidationErrs, err := v.validateWorkerZoneChanges(field.NewPath("spec", "providers", "workers"), shoot, seedTemplate)
+	if err != nil {
+		return apierrors.NewInternalError(err)
+	}
+	allErrs = append(allErrs, zoneValidationErrrs...)
```

Comment:
Sorry for not providing a complete suggestion in the previous comment.

```suggestion
	allErrs = append(allErrs, zoneValidationErrs...)
```